src/interaction/sessionrecording.cpp

```diff
@@ -0,0 +1,1207 @@
+            seekAheadIndex = static_cast<unsigned int>(_timeline.size()) - 1;
+
+        if (doesTimelineEntryContainCamera(seekAheadIndex)) {
+            unsigned int indexIntoCameraKeyframes
+                = _timeline[seekAheadIndex].idxIntoKeyframeTypeArray;
+            double seekAheadKeyframeTimestamp = _timeline[seekAheadIndex].timestamp;
+
+            if (indexIntoCameraKeyframes >= (_keyframesCamera.size() - 1))
+                _hasHitEndOfCameraKeyframes = true;
+
+            if (currTime < seekAheadKeyframeTimestamp) {
+                if (seekAheadIndex > _idxTimeline_cameraPtrNext) {
+                    _idxTimeline_cameraPtrPrev = _idxTimeline_cameraPtrNext;
+                    _idxTimeline_cameraPtrNext = seekAheadIndex;
+                }
+                break;
+            } else {
+                //Force interpolation between consecutive keyframes
+                _idxTimeline_cameraPtrPrev = seekAheadIndex;
+            }
+        }
+
+        double interpolationUpperBoundTimestamp
+            = _timeline[_idxTimeline_cameraPtrNext].timestamp;
+        if ((currTime > interpolationUpperBoundTimestamp) && _hasHitEndOfCameraKeyframes) {
+            _idxTimeline_cameraPtrPrev = _idxTimeline_cameraPtrNext;
+            return false;
+            break;
+        }
+
+        if (seekAheadIndex == (_timeline.size() - 1))
+            break;
+    }
+    return true;
+}
+
+bool SessionRecording::doesTimelineEntryContainCamera(unsigned int index) {
+    return (_timeline[index].keyframeType == recordedType::camera);
+}
+
+bool SessionRecording::processNextNonCameraKeyframeAheadInTime() {
+    bool returnValue = false;
+
+    //LINFO(fmt::format("Keyframe at {} frame={} timelineIndex={}", now, global::renderEngine._frameNumber, _idxTimeline));
+
+    switch (getNextKeyframeType()) {
+    case recordedType::camera:
+        //Just return true since this function no longer handles camera keyframes
+        returnValue = true;
+        break;
+
+    case recordedType::time:
+        _idxTime = _timeline[_idxTimeline_nonCamera].idxIntoKeyframeTypeArray;
+        if (_keyframesTime.size() == 0) {
+            return false;
+        }
+        LINFO("Time keyframe type");
+        //TBD: the TimeManager restricts setting time directly
+        break;
+
+    case recordedType::script:
+        _idxScript = _timeline[_idxTimeline_nonCamera].idxIntoKeyframeTypeArray;
+        returnValue = processScriptKeyframe();
+        break;
+
+    default:
+        LERROR(fmt::format("Bad keyframe type encountered during playback at index {}.",
+            _idxTimeline_nonCamera));
+        break;
+    }
+    return returnValue;
+}
+
+//void SessionRecording::moveBackInTime() { } //for future use
+
+unsigned int SessionRecording::findIndexOfLastCameraKeyframeInTimeline() {
+    unsigned int i;
+    for (i = static_cast<unsigned int>(_timeline.size()) - 1; i > 0; i--) {
```

Comment:
Can move the `unsigned int i` defintiion into the for loop and instead of `breaking`, just returning the value

Reply:
Made a different change, for the rare case of no camera keyframe found, would still return a valid index.